src/main/java/com/facebook/openwifirrm/ucentral/UCentralClient.java

```diff
@@ -270,6 +367,7 @@ public boolean isInitialized() {
 		) {
 			return false;
 		}
+		accessToken = getAccessToken();
```

Comment:
The important part of this diff is where you are calling `getAccessToken()`. This seems like a pretty random spot, and even though it probably works, it is totally unexpected for an "is initialized?" method to do this.

Was your intention to check expiry at the start of each thread's event loop? That approach seems okay, but it should just be a separate/explicit function call. The alternative is replacing *all* direct usage of `accessToken` with this function, but that is slightly more complicated and possibly introduces strange timing/delays.

Other problems with this:
- Access token is only used if `usePublicEndpoints` is true, so don't call this unless necessary
- This is not thread safe - in fact it is very likely that two threads will try to refresh the token at the same time. You need to use some form of synchronization to ensure that only 1 refresh is attempted at a time (and in an efficient way that doesn't involve blocking other threads unnecessarily)

Reply:
Yes the intention is that as `isInitialized` is called at the start of some events, checking the expiry at every implementation running. What do you mean by a separate function call? Can I make it as a function and call it at each thread's event loop as where `isInitialized` is called except putting it directly in `isInitialized`?

For the alternative, I have replaced usage of `accessToken` with `getAccessToken()` except login and somewhere it would be falling into infinite loop(For example, `httpPost` when sending token refresh request). Do you think it's a good idea to check the expiry In every httpGet/httpPost? 

It makes more sense to me checking expiry in every httpGet/httpPost, can we hold the both to ensure as long as the access token expired, it is refreshed immediately?

Reply:
Your approach is better I think. What I mean is you should write another function like `refreshAccessToken()` to call *after* the `isInitialized()` loop because a function like "is initialized" implies you aren't doing any actions like this.